### PR TITLE
glib, gnutls and libgcrypt security updates

### PR DESCRIPTION
**Pull request by th0ma7**
_Motivation:_  While investigating issue #4006 and trying to find the root cause I ended-up updating various dependencies.  In the process I noticed multiple old versions of various key libraries that needed to be updated due to multiple CVE advisories.  This PR is in direct relation to #4155 where newer version addresses security issues.
_Linked issues:_  #4006, #4008, #3998, #4155 

### Checklist
- [x] Build rule `all-supported` completed successfully
- n/a Package upgrade completed successfully
- n/a New installation of package completed successfully

## IMPORTANT NOTES
- This PR applies on top of #4155 
- Current github-action build failures should normally disappear once other PR are merged.

### GLIB
https://www.cvedetails.com/vulnerability-list/vendor_id-283/product_id-16275/version_id-260024/Gnome-Glib-2.56.1.html
- `glib` update from 2.55 to 2.58 (was using an old and unstable version using odd versioning)
- `glibmm` update from 2.34 to 2.58 (should always be in sync with `glib` in terms of versioning)
- `glib` and `glibmm` version 2.58.x is the last version supporting autoconf/autobuild tools.  Starting with version 2.60 only meson/ninja are supported
### gnutls & libgcrypt
https://www.cvedetails.com/vulnerability-list/vendor_id-72/product_id-4433/GNU-Gnutls.html
https://www.cvedetails.com/vulnerability-list/vendor_id-4711/product_id-25777/Gnupg-Libgcrypt.html
- gnutls: Update from version 3.6.8 to 3.6.15
- libgcrypt: Update from version 1.8.5 to 1.8.6
- libgpg-error: Update from version 1.32 to 1.39 
### Needed to be updated to accommodate above version updates
- `libsigc++` update from 2.4.0 to 2.10
- `libxml++` update from 2.36 to 2.40.1
### Others
- `rmlint` & `jpcre2` are `synocli-file` dependencies to which trivial updates where available (relates on PR #4155)
-- `jpcre2` update from version 10.31.03 to 10.32.01
-- `rmlint` update from 2.9 to 2.10.1
- `sslh` Update from 1.19c to 1.21c
- `gnupg` update from 2.1.6 to 2.2.23
-  `libassuan` Update from 2.2.1 to 2.5.3 (for `gnupg`)
- `libksba` Update from version 1.3.3 to 1.4.0 (for `gnupg`)

### CODE REVIEW

**Comment by th0ma7** [comment]
I don't know if there is interest in this; instead of loosing the work I thought I would share.

Many packages dated quite a bit or where based on unstable releases (e.g. `glib`) or misaligned (`glibmm`).  Feel free to comment or disregard as you see fit.

File: native/glib/Makefile [hunk 1]
```diff
@@ -14,4 +14,14 @@ LICENSE  = GPLv2
...
+
 include ../../mk/spksrc.native-cc.mk
+
+.PHONY: glib_pre_configure
+
+glib_pre_configure: 
+	# force autogen to first use internal pcre instead of provided to prevent
+	# build failure.  Subsequent default configure will superseed this afterwards
+	$(RUN) ./autogen.sh
+#	$(RUN) ./autogen.sh --with-pcre
```
Comment (ymartin59) [inline native/glib/Makefile:27]: I would discard this commented line

**Review by ymartin59 — APPROVED** [review]

File: native/glib/Makefile (hunk 1 above)
Comment (th0ma7) [reply native/glib/Makefile:27]: done.

**Comment by th0ma7** [comment]
@ymartin59 Now ready for a merge.  Feel free to have another look at it before I squash & merge but should now build with no more errors.

**Review by ymartin59 — APPROVED** [review]

**Pull request merged**